compiler/src/typed/module_resolution.re

```
@@ -276,46 +276,48 @@ module Dependency_graph =
+                  dn_latest_resolution: ref(Some(located)),
+                };
+              };
+            },
+            Grain_parsing.Driver.scan_for_imports(srcpath),
+          );
+        };
+
         // TODO: (#597) Propagating the compiler flag information correctly is tricky.
         //        For the moment, from the dependency graph's perspective, we assume that
         //        nothing uses --no-pervasives or --no-gc.
         switch (located) {
         | None => failwith("get_dependencies: Should be impossible")
         | Some(WasmModule(_)) => []
-        | Some(GrainModule(srcpath, None)) =>
-          let ret =
-            List.map(
-              name => {
-                let located =
-                  locate_module(base_dir, active_search_path, name);
-                let out_file_name = located_to_out_file_name(located);
-                let existing_dependency = lookup(out_file_name);
-                switch (existing_dependency) {
-                | Some(ed) =>
-                  Hashtbl.add(ed.dn_unit_name, Some(dn), name);
-                  ed;
-                | None =>
-                  let tbl = Hashtbl.create(8);
-                  Hashtbl.add(tbl, Some(dn), name);
-                  {
-                    dn_unit_name: tbl,
-                    dn_file_name: out_file_name,
-                    dn_up_to_date: ref(false), // <- needs to be checked
-                    dn_latest_resolution: ref(Some(located)),
-                  };
-                };
-              },
-              Grain_parsing.Driver.scan_for_imports(srcpath),
-            );
-          ret;
+        | Some(GrainModule(srcpath, None)) => from_srcpath(srcpath)
         | Some(GrainModule(srcpath, Some(objpath))) =>
-          let cmi = read_file_cmi(objpath);
-          let ret =
+          switch (read_file_cmi(objpath)) {
+          | exception (Cmi_format.Error(_)) => from_srcpath(srcpath)
```

Just wondering should we maybe inline comments next to code like this? Normally, I would see something along these lines and think we're erroneously swallowing an error.

---

Added a comment!